src/coprocessor/dag/expr/compare.rs

```diff
@@ -157,6 +158,16 @@ impl FnCall {
     ) -> Result<Option<Cow<'a, Json>>> {
         do_coalesce(self, |v| v.eval_json(ctx, row))
     }
+
+    pub fn like(&self, ctx: &StatementContext, row: &[Datum]) -> Result<Option<i64>> {
+        let target = try_opt!(self.children[0].eval_string_and_decode(ctx, row));
```

Comment:
Does target have to be a string? Can it be a blob?

Reply:
If call LIKE on `blob`, TiDB type infer mechanism will convert that to `String`. Now I don't know how mysql convets `blob` to `string`. Both TiDB and TiKV treats those bytes as **UTF8**. I will figure that.

Reply:
If we can't treat blob bytes as UTF8, I think we should support the real codec in `eval_string_and_decode` and call that function in `cast_string_as_string`. Maybe raise a new PR is better?